src/Scripts/Ecs/Core/Entity.gd

```diff
@@ -21,4 +27,8 @@ func remove(type:String):
 		var component = self._components[type]
 		component.parent = null
 		self._components.erase(type)
-	
\ No newline at end of file
+	
+############ extensions
+func sprite(image:String, layer:String):
+	self._components["SpriteComponent"] = SpriteComponent.new(image, layer)
+	return self
```

Comment:
Does it make sense to have a method in the entity class for each component? I suppose this is meant to be a static extension kind of thing, and if so, it does make sense, but since gdscript doesn't support static extensions afaik, we have to be careful lest we end up with a blob entity.

Moreover, shouldn't this method internally use `Entity.add`? It appears otherwise unused.

Reply:
Correct on both counts. This is just a "fluent API" that makes it easier to use. We can mark the region for it so we can extract if it gets too blobby.

Yeah, it should use `Entity.add` - good catch.